hal/src/sercom/v2/uart/mod.rs

```diff
@@ -0,0 +1,1528 @@
+    }
+
+    /// Read the interrupt flags
+    pub fn read_flags(&self) -> Flags {
+        let bits = self.usart().intflag.read().bits();
+        Flags::from_bits_truncate(bits)
+    }
+
+    /// Clear interrupt status flags
+    ///
+    /// Setting the `ERROR`, `RXBRK`, `CTSIC`, `RXS`, or `TXC` flag will clear
+    /// the interrupts. This function has no effect on the `DRE` or
+    /// `RXC` flags.
+    ///
+    /// Note that only the flags pertinent to `Self`'s [`Capability`]
+    /// will be cleared. The other flags will be **SILENTLY IGNORED**.
+    ///
+    /// * Available flags for [`Receive`] capability: `RXC`, `RXS`, `RXBRK` and
+    ///   `ERROR`
+    /// * Available flags for [`Transmit`] capability: `DRE`, `TXC` and `CTSIC`
+    /// * Since [`Duplex`] [`Uart`]s are [`Receive`] + [`Transmit`] they have
+    ///   all flags available.
+    pub const TX_FLAG_MASK: u8 = DRE | TXC | CTSIC;
+    ///
+    /// **Warning:** The implementation of of [`Write::flush`] waits on and
+    /// clears the `TXC` flag. Manually clearing this flag could cause it to
+    /// hang indefinitely.
+    pub fn clear_flags(&mut self, flags: Flags) {
+        // Remove flags not pertinent to Self's Capability
+        self.usart()
+            .intflag
+            .modify(|_, w| unsafe { w.bits(Self::capability_flags(flags).bits()) });
+    }
+
+    /// Enable interrupts for the specified flags.
+    ///
+    /// Note that only the flags pertinent to `Self`'s [`Capability`]
+    /// will be cleared. The other flags will be **SILENTLY IGNORED**.
+    ///
+    /// * Available flags for [`Receive`] capability: `RXC`, `RXS`, `RXBRK` and
+    ///   `ERROR`
+    /// * Available flags for [`Transmit`] capability: `DRE`, `TXC` and `CTSIC`
+    /// * Since [`Duplex`] [`Uart`]s are [`Receive`] + [`Transmit`] they have
+    ///   all flags available.
+    #[inline]
+    pub fn enable_interrupts(&mut self, flags: Flags) {
+        self.usart()
+            .intenset
+            .write(|w| unsafe { w.bits(Self::capability_flags(flags).bits()) });
+    }
+
+    /// Disable interrupts for the specified flags.
+    ///
+    /// Note that only the flags pertinent to `Self`'s [`Capability`]
+    /// will be cleared. The other flags will be **SILENTLY IGNORED**    
+    ///
+    /// * Available flags for [`Receive`] capability: `RXC`, `RXS`, `RXBRK` and
+    ///   `ERROR`
+    /// * Available flags for [`Transmit`] capability: `DRE`, `TXC` and `CTSIC`
+    /// * Since [`Duplex`] [`Uart`]s are [`Receive`] + [`Transmit`] they have
+    ///   all flags available.
+    pub fn disable_interrupts(&mut self, flags: Flags) {
+        self.usart()
+            .intenclr
+            .write(|w| unsafe { w.bits(Self::capability_flags(flags).bits()) });
+    }
+
+    /// Read the status flags
+    #[inline]
+    pub fn read_status(&self) -> Status {
+        let bits = self.usart().status.read().bits();
+        Status::from_bits_truncate(bits)
+    }
```

Comment:
By the way, I renamed my `Errors` type to be `Status` and renamed my corresponding functions. You were right to stick with `Status`. Having both `Error` and `Errors` in the same module is confusing.